`TestDumpBinaryTime` failed in Egypt time

## Bug Report

### 1. Minimal reproduce step (Required)

➜  tidb git:(master) ✗ TZ="Africa/Cairo" go test --tags=intest -timeout 30s -run ^TestDumpBinaryTime$ "github.com/pingcap/tidb/pkg/server/internal/dump"

### 2. What did you expect to see? (Required)
Success

### 3. What did you see instead (Required)
--- FAIL: TestDumpBinaryTime (0.00s)
    dump_test.go:33:
        	Error Trace:	/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/server/internal/dump/dump_test.go:33
        	Error:      	Received unexpected error:
        	            	[types:1292]Incorrect time value: '{1991 5 1 1 1 1 100001}'
        	            	github.com/pingcap/errors.AddStack
        	            		/Users/bb7133/Projects/gopath/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20240318064555-6bd07397691f/errors.go:178
        	            	github.com/pingcap/errors.(*Error).GenWithStackByArgs
        	            		/Users/bb7133/Projects/gopath/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20240318064555-6bd07397691f/normalize.go:175
        	            	github.com/pingcap/tidb/pkg/types.CoreTime.GoTime
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/types/core_time.go:182
        	            	github.com/pingcap/tidb/pkg/types.(*Time).ConvertTimeZone
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/types/time.go:365
        	            	github.com/pingcap/tidb/pkg/types.checkTimestampType
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/types/time.go:2172
        	            	github.com/pingcap/tidb/pkg/types.Time.Check
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/types/time.go:685
        	            	github.com/pingcap/tidb/pkg/types.parseTime
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/types/time.go:2012
        	            	github.com/pingcap/tidb/pkg/types.ParseTime
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/types/time.go:1988
        	            	github.com/pingcap/tidb/pkg/types.ParseTimestamp
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/types/time.go:2025
        	            	github.com/pingcap/tidb/pkg/server/internal/dump.TestDumpBinaryTime
        	            		/Users/bb7133/Projects/gopath/src/github.com/pingcap/tidb/pkg/server/internal/dump/dump_test.go:32
        	            	testing.tRunner
        	            		/Users/bb7133/Softwares/go/src/testing/testing.go:1595
        	            	runtime.goexit
        	            		/Users/bb7133/Softwares/go/src/runtime/asm_arm64.s:1197
        	Test:       	TestDumpBinaryTime
FAIL
FAIL	github.com/pingcap/tidb/pkg/server/internal/dump	0.292s

"1991-05-01 01:01:01.100001" is not a valid time in "Africa/Cairo" timezone because of the [DST change](https://www.timeanddate.com/time/change/egypt?year=1991):

> Wednesday, May 1, 1991, 1:00:00 am clocks were turned forward 1 hour to
> Wednesday, May 1, 1991, 2:00:00 am local daylight time instead.

### 4. What is your TiDB version? (Required)

It doesn't matter.

--- a/pkg/server/internal/dump/dump_test.go
+++ b/pkg/server/internal/dump/dump_test.go
@@ -29,7 +29,7 @@ func TestDumpBinaryTime(t *testing.T) {
 	d := BinaryDateTime(nil, parsedTime)
 	require.Equal(t, []byte{0}, d)
 
-	parsedTime, err = types.ParseTimestamp(typeCtx.WithLocation(time.Local), "1991-05-01 01:01:01.100001")
+	parsedTime, err = types.ParseTimestamp(typeCtx.WithLocation(time.UTC), "1991-05-01 01:01:01.100001")
 	require.NoError(t, err)
 	d = BinaryDateTime(nil, parsedTime)
 	// 199 & 7 composed to uint16 1991 (litter-endian)

